simulariumio/data_objects/agent_data.py

```diff
@@ -330,6 +330,90 @@ def from_buffer_data(
             draw_fiber_points=False,
         )
 
+    @staticmethod
+    def fill_df(df, fill):
+        """
+        Fill Nones in a DataFrame with a fill value
+        """
+        # Create a dataframe of fill values
+        fill_array = [[fill] * df.shape[1]] * df.shape[0]
+        fill_df = pd.DataFrame(fill_array)
+        # Replace all entries with None with the fill
+        df[df.isna()] = fill_df
+        return df
+
+    @staticmethod
+    def jagged_3d_list_to_numpy_array(jagged_3d_list):
+        """
+        Shape a jagged list with 3 dimensions to a numpy array
+        """
+        df = AgentData.fill_df(pd.DataFrame(jagged_3d_list), [0, 0, 0])
+        df_t = df.transpose()
+        exploded = [df_t[col].explode() for col in list(df_t.columns)]
+        return np.array(exploded, dtype=float).reshape((df.shape[0], df.shape[1], 3))
+
+    @staticmethod
+    def get_subpoints_numpy_array(trajectory) -> np.ndarray:
```

Comment:
for all of these methods (other than `from_lists()`), they can be private so prepend `_` to the func name, this way there will be less clutter in the auto generated docs